### PR TITLE
Add new memory runLog info to memoryProfiler.py 

### PR DESCRIPTION
**Pull request by albeanth**
## What is the change?
Adds some potentially useful `runLog` statements to the memory profiler interface on how much memory is is use , how much is available, and the total available for a simulation. The first two are dynamic (i.e., change with (cycle, node)) while the third is static. 

## Why is the change being made?
Some memory intensive downstream analysis found this useful, so we thought we'd push it up to ARMI. 

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [x] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify any new/changed code.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/developer/tooling.html#add-release-notes) have been updated if necessary.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

File: armi/bookkeeping/memoryProfiler.py [hunk 1]
```diff
@@ -78,6 +114,11 @@ def __init__(self, r, cs):
 
     def interactBOL(self):
         interfaces.Interface.interactBOL(self)
+
+        # prints the bottom line memory state...
+        # the code after this is somewhat confusing to interpret
+        printCurrentMemoryState(self.cs["mpiTasksPerNode"])
```
Comment (albeanth) [inline armi/bookkeeping/memoryProfiler.py:120]: the new equivalent of this setting isn't available in ARMI anymore.... shoot.

Comment (albeanth) [reply armi/bookkeeping/memoryProfiler.py]: after chatting with @opotowsky we put a bandaid `try/except` on this

File: armi/bookkeeping/memoryProfiler.py [hunk 2]
```diff
@@ -36,16 +36,19 @@
 https://pythonhosted.org/psutil/
 https://docs.python.org/3/library/gc.html#gc.garbage
 """
+from os import cpu_count
 from typing import Optional
 import gc
 import sys
+from math import floor
```
Comment (john-science) [inline armi/bookkeeping/memoryProfiler.py:43]:
```suggestion
from math import floor
from os import cpu_count
from typing import Optional
import gc
import sys
```

Comment (john-science) [reply armi/bookkeeping/memoryProfiler.py:43]: just alphabetizing the imports

File: armi/bookkeeping/memoryProfiler.py [hunk 3]
```diff
@@ -101,11 +129,30 @@ def interactEveryNode(self, cycle, node):
             mpiAction.broadcast().invoke(self.o, self.r, self.cs)
 
     def interactEOL(self):
-        r"""End of life hook. Good place to wrap up or print out summary outputs."""
+        """End of life hook. Good place to wrap up or print out summary outputs."""
         if self.cs["debugMem"]:
```
Comment (john-science) [inline armi/bookkeeping/memoryProfiler.py:133]: I'm just wondering... should this same settings check be done at `interactBOL` and `interactEveryNode`?

I could be totally off-base here, I just can't tell why this is only done at the one time node.

**Review by john-science — APPROVED** [review]
Approved! I have a couple of minor thoughts for you, but no hills I'm prepared to die on.

File: armi/bookkeeping/memoryProfiler.py (hunk 3 above)
Comment (albeanth) [reply armi/bookkeeping/memoryProfiler.py:133]: I'm not following. Can you elaborate?

Comment (john-science) [reply armi/bookkeeping/memoryProfiler.py:133]: Well, this memory profiler does nothing in `interactEOL` unless the setting `debugMem` is set to `True`.

So.... I would EXPECT that setting would control all the time points, not just the one. 

If that's not the case cool, I'm just wondering what the difference is.

Comment (albeanth) [reply armi/bookkeeping/memoryProfiler.py:133]: That setting does get used in `interactBOL` and `interactEveryNode`. These two just also do other stuff too. 

Granted I don't know the original intent. The git blame for the vast majority of this module go well beyond my time with TerraPower.

**Review by opotowsky — APPROVED** [review]
I'm a fan! 

Note that we realize this code probably shouldn't be in ARMI but to push towards a deadline we have [a separate cleanup ticket](https://github.com/terrapower/armi/issues/1975) to bring parts of the memory profiler into internal tooling, where it belongs.

**Pull request merged**